sdk/spring/azure-spring-boot/src/test/java/com/azure/spring/aad/implementation/AzureActiveDirectoryConfigurationTest.java

```diff
@@ -163,6 +163,38 @@ public void customizeUri() {
         assertEquals(endpoints.jwkSetEndpoint("fake-tenant-id"), azure.getProviderDetails().getJwkSetUri());
     }
 
+    @Test
+    public void clientRequiresOnDemandPermissions() {
+        AnnotationConfigApplicationContext context = getContext();
+        TestPropertySourceUtils.addInlinedPropertiesToEnvironment(context,
+            "azure.activedirectory.authorization.graph.scopes = Calendars.Read",
+            "azure.activedirectory.authorization.graph.on-demand = true",
+            "azure.activedirectory.authorization.arm.scopes = https://management.core.windows.net/user_impersonation"
+        );
+        context.refresh();
+
+        AzureClientRegistrationRepository repo = context.getBean(AzureClientRegistrationRepository.class);
+        ClientRegistration azure = repo.findByRegistrationId("azure");
+        ClientRegistration graph = repo.findByRegistrationId("graph");
+        ClientRegistration arm = repo.findByRegistrationId("arm");
+
+        assertNotNull(azure);
+        assertDefaultScopes(
+            azure,
+            "openid",
+            "profile",
+            "https://graph.microsoft.com/User.Read",
```

Comment:
`https://graph.microsoft.com/User.Read` belong to `graph` client, so `https://graph.microsoft.com/User.Read` should not exist in `DefaultScopes`, is that right?

Reply:
`https://graph.microsoft.com/User.Read` belong to default client if allowedGroups is not null. 
https://github.com/Azure/azure-sdk-for-java/blob/d1d7fe0c1dbfa4a6b1c157cbe6c07cdfa59efab2/sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/implementation/AzureActiveDirectoryConfiguration.java#L77

Reply:
It make sense.